config/config.go

```diff
@@ -217,7 +216,7 @@ type ScrapeConfig struct {
 	// Indicator whether the scraped metrics should remain unmodified.
 	HonorLabels bool `yaml:"honor_labels,omitempty"`
 	// A set of query parameters with which the target is scraped.
-	Params url.Values `yaml:"params,omitempty"`
+	Params map[string]string `yaml:"params,omitempty"`
```

Comment:
It would have to be configureable. We face the same problem anywhere else when concatenating. In practice it's usually trivial to find a char that's not used with high certainty as you know your data.

Introducing the required config option (think default value separator) comes with some implementation complexity.
Numbering has less for us. In exchange the user has to carry the burden of order and potentially very verbose relabeling configs (has to replicate for each numbered parameter).

I reckon usage of query params could grow in the future so use cases may grow.

As usual, more opinions here would help a lot.